src/components/stepforms/JoinStepForm.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div>
+    <step-form-title :title="title" />
+    <AutocompleteWidget
+      id="rightPipelineInput"
+      v-model="editedStep.right_pipeline"
+      name="Select a dataset to join (as right dataset):"
+      :options="Object.keys(pipelines).filter(p => p !== currentPipelineName)"
```

Comment:
side-note: this is a detail and there's no need to handle this now but there are valid cases where we want to join a dataset with itself.